comp/api/api/apiimpl/internal/agent/agent.go

```diff
@@ -87,11 +89,16 @@ func SetupHandlers(
 	collector optional.Option[collector.Component],
 	eventPlatformReceiver eventplatformreceiver.Component,
 	ac autodiscovery.Component,
+	providers []api.AgentEndpointProvider,
 ) *mux.Router {
 
+	// TODO: move these to a component that is registerable
+	for _, p := range providers {
+		r.Handle(p.Provider.Route(), p.Provider).Methods(p.Provider.Method())
```

Comment:
of note this is now a `http.Handle` call so the provided endpoints need to respond to `ServeHTTP(http.ResponseWriter, *http.Request)` which I think is standard

I saw promhttp would return a HandleFunc for the `comp/core/telemetry` componenet
https://github.com/DataDog/datadog-agent/blob/e4d347d1aaf019a24845b62da13ec66d815738c1/comp/core/telemetry/telemetry.go#L72-L74

Is there a preference between the two approaches?